Modules/mmapmodule.c

```diff
@@ -682,6 +682,60 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            Py_UNREACHABLE();
+    }
+
+    const char *tp_name = Py_TYPE(self)->tp_name;
+
+#ifdef MS_WINDOWS
+    if (m_obj->map_handle == NULL)
+#elif defined(UNIX)
+    if (m_obj->data == NULL)
+#endif
+    {
+        reprfmt = "<%s closed=True access=%s>";
+        repr = PyUnicode_FromFormat(reprfmt, tp_name, access_str);
```

Comment:
Even for closed one, I think `offset` and `length` could still be displayed, they are passed in by users and not gonna be changed. We could tell more info then.

Reply:
In fact they could get changed: https://github.com/python/cpython/blob/master/Modules/mmapmodule.c#L557

Reply:
Doesn't matter. Once it's closed, you can't do more operations on it. Just show what they are when they are closed.